pkg/relayer/config/types.go

```diff
@@ -0,0 +1,141 @@
+// supplier is of type "http"
+type YAMLRelayMinerSupplierServiceAuthentication struct {
+	Username string `yaml:"username,omitempty"`
+	Password string `yaml:"password,omitempty"`
+}
+
+// RelayMinerConfig is the structure describing the RelayMiner config
+type RelayMinerConfig struct {
+	PocketNode     *RelayMinerPocketNodeConfig
+	SigningKeyName string
+	SmtStorePath   string
+	Proxies        map[string]*RelayMinerProxyConfig
+}
+
+// RelayMinerPocketNodeConfig is the structure resulting from parsing the pocket
+// node URLs section of the RelayMiner config file
+type RelayMinerPocketNodeConfig struct {
+	QueryNodeRPCUrl  *url.URL
+	QueryNodeGRPCUrl *url.URL
+	TxNodeGRPCUrl    *url.URL
+}
+
+// RelayMinerProxyConfig is the structure resulting from parsing the proxy
+// section of the RelayMiner config file.
+// Each proxy embeds a map of supplier configs that are associated with it.
+// Other proxy types may embed other fields in the future. eg. "https" may
+// embed a TLS config.
+type RelayMinerProxyConfig struct {
+	// Name is the name of the proxy server, used to identify it in the config
+	Name string
+	// Type is the transport protocol used by the proxy server like (http, https, etc.)
+	Type ProxyType
+	// Host is the host on which the proxy server will listen for incoming
+	// relay requests
+	Host string
+	// XForwardedHostLookup is a flag that indicates whether the proxy server
+	// should lookup the host from the X-Forwarded-Host header before falling
+	// back to the Host header.
+	XForwardedHostLookup bool
+	// Suppliers is a map of serviceIds -> RelayMinerSupplierConfig
+	Suppliers map[string]*RelayMinerSupplierConfig
+}
+
+// RelayMinerSupplierConfig is the structure resulting from parsing the supplier
+// section of the RelayMiner config file.
+type RelayMinerSupplierConfig struct {
+	// Name is the serviceId corresponding to the current configuration.
+	Name string
+	// Type is the transport protocol used by the supplier, it must match the
+	// type of the proxy it is associated with.
+	Type ProxyType
+	// Hosts is a list of hosts advertised on-chain by the supplier, the corresponding
+	// proxy server will accept relay requests for these hosts.
+	Hosts []string
+	// ServiceConfig is the config of the service that relays will be proxied to.
+	// Other supplier types may embed other fields in the future. eg. "https" may
+	// embed a TLS config.
+	ServiceConfig *RelayMinerSupplierServiceConfig
+}
+
+// RelayMinerSupplierServiceConfig is the structure resulting from parsing the supplier
+// service sub-section of the RelayMiner config file.
+type RelayMinerSupplierServiceConfig struct {
+	// Url is the URL of the service that relays will be proxied to.
+	Url *url.URL
+	// Authentication is the basic auth structure used to authenticate to the
+	// request being proxied from the current proxy server.
+	// If `RelayMinerSupplierConfig.Type` is `ProxyTypeHTTP` then `Authentication`
```

Comment:
Whe we go type specific, can you add a `For example`. Ditto below

Reply:
Good catch! I confused it with proxy type related fields. This is independent of the proxy type indeed.